packages/TorneloScoresheet/__tests__/useAppModeState.test.ts

```diff
@@ -32,4 +32,53 @@ describe('useAppModeState', () => {
       view: ArbiterModeViews.EnterPgnLink,
     });
   });
+
+  const pgnUrl =
+    'https://staging-api.tornelo.com/divisions/a1aaede1-fba1-4d63-a92b-acb789b9dcce/broadcast-pgn?convertMatchNumbers=false&includeDivisionName=true&round=6&token=46260fc4ed6887bbb45e8f793a70b950&withClocks=true';
+  test('enterTablePairingSelectionViewValidUrl', async () => {
+    const { result } = renderHook(() => useAppModeState(), {
+      wrapper: AppModeStateContextProvider,
+    });
+    // Valid url, should go to tableparing selection
+    await act(async () => {
+      const enterPairingModeResult =
+        await result.current[1].goToTablePairingSelection(pgnUrl);
+      expect(isError(enterPairingModeResult)).toEqual(false);
+    });
+
+    // ensure correct mode
+    expect(result.current[0].mode).toEqual(AppMode.ArbiterSetup);
+    if (result.current[0].mode === AppMode.ArbiterSetup) {
+      // ensure correct view
+      expect(result.current[0].view).toEqual(
+        ArbiterModeViews.TablePairingSelection,
+      );
+
+      // ensure all pairings parsed
+      expect(result.current[0].pairings).toBeDefined();
+      if (result.current[0].pairings) {
+        expect(result.current[0].pairings.length).toEqual(7);
+      }
+    }
+  });
+
+  test('goBackToPgnEnterView', async () => {
+    const { result } = renderHook(() => useAppModeState(), {
+      wrapper: AppModeStateContextProvider,
+    });
+
+    // go into table pairing selection view
+    await act(async () => {
+      await result.current[1].goToTablePairingSelection(pgnUrl);
+    });
+
+    // go back to pgn enter view
+    result.current[1].goToEnterPgnLink();
```

Comment:
I think this should be in a call to `act`? This is causing the test to fail I think